packages/utils/src/normalize.ts

```diff
@@ -51,170 +53,173 @@ export function normalizeToSize<T>(
 }
 
 /**
- * Walks an object to perform a normalization on it
+ * Visits a node to perform a normalization on it
  *
- * @param key of object that's walked in current iteration
- * @param value object to be walked
+ * @param key The key corresponding to the given node
+ * @param value The node to be visited
  * @param depth Optional number indicating how deep should walking be performed
- * @param maxProperties Optional maximum  number of properties/elements included in any single object/array
+ * @param maxProperties Optional maximum number of properties/elements included in any single object/array
  * @param memo Optional Memo class handling decycling
  */
-export function walk(
+export function visit(
   key: string,
-  value: UnknownMaybeWithToJson,
+  value: unknown,
   depth: number = +Infinity,
   maxProperties: number = +Infinity,
   memo: MemoFunc = memoBuilder(),
-): unknown {
+): Primitive | { [key: string]: unknown } {
   const [memoize, unmemoize] = memo;
 
-  // If we reach the maximum depth, serialize whatever is left
-  if (depth === 0) {
-    return serializeValue(value);
+  // if the value has a `toJSON` method, bail and let it do the work
+  const valueWithToJSON = value as unknown & { toJSON?: () => string };
+  if (valueWithToJSON && typeof valueWithToJSON.toJSON === 'function') {
```

Comment:
Hmmm. I may be misunderstanding you, but a) my typescript isn't screaming, and b) I claim that's because checking the truthiness of `valueWithToJson` protects against just the case you're raising. No?

Reply:
You're totally right. My bad. For some stupid reason I didn't catch that truthiness check.

By "ts is screaming" I meant, that it's screaming if I remove the type cast - which I now realize was stupid of me :D